### ISSUE TITLE
Tying to use Payload local api in nextjs pages router api route breaks route.

### ISSUE THREAD

**Issue by ckruppe** [payloadcms/payload#7640]
### Payload Version

3.0.0-beta.77

### Node Version

20.11.1

### Next.js Version

15.0.0-rc.0

### Describe the Bug

If i try to use the Payload local api ass described here: https://payloadcms.com/docs/beta/local-api/overview#importing-it the server breaks with the following message:
```
TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".scss" for C:\projekte\Intern\Pages\coding-cards\node_modules\@payloadcms\richtext-slate\dist\field\buttons.scss
    at Object.getFileProtocolModuleFormat [as file:] (node:internal/modules/esm/get_format:160:9)
    at defaultGetFormat (node:internal/modules/esm/get_format:203:36)
    at defaultLoad (node:internal/modules/esm/load:143:22)
    at async ModuleLoader.load (node:internal/modules/esm/loader:409:7)
    at async ModuleLoader.moduleProvider (node:internal/modules/esm/loader:291:45) {
  code: 'ERR_UNKNOWN_FILE_EXTENSION'
}
```

an scss loader is already injected by nextjs config but it seems nextjs does not use it in its server compile pass. Maybe i do also something wrong here. The the simple example of my controller class that gets called by the nextjs api route handler is this:

```typescript
import {type BotBuilderCloudAdapter, NotificationTargetType} from '@microsoft/teamsfx';
import {getPayloadHMR} from '@payloadcms/next/utilities';
import config from '@payload-config';
import {conversationBotApp} from 'Utils/conversationBotApp';

export class CardService {
    static async filterMembersByRecipients(members: BotBuilderCloudAdapter.Member[]) {
        const payload = await getPayloadHMR({config});
        const recipients = await payload.find({
            collection: 'recipients',
            depth: 2,
            limit: 500
        });

        console.log(recipients);
        console.log(members);

        return [];
    }
}
```

### Reproduction Steps

As i have seen its enough to try to import the config eather in an api route or an page with getServerSideProps.
On both occourances i get the same error.

**Comment by jmikrut** [comment]
Hey @ckruppe — this will be resolved by an incoming PR here:

https://github.com/payloadcms/payload/pull/7620

Once this is merged (hopefully tomorrow) this will be fixed! Keep an eye out!

**Comment by ckruppe** [comment]
@jmikrut Hey thank you for the fast reply. I could remedy the direct issue with this conig in next.config.js:
`transpilePackages: ['@payloadcms/richtext-slate']`

But got me only to the next error:
```
TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".css" for C:\projekte\Intern\Pages\coding-cards\node_modules\react-image-crop\dist\ReactCrop.css
    at Object.getFileProtocolModuleFormat [as file:] (node:internal/modules/esm/get_format:160:9)
    at defaultGetFormat (node:internal/modules/esm/get_format:203:36)
    at defaultLoad (node:internal/modules/esm/load:143:22)
    at async ModuleLoader.load (node:internal/modules/esm/loader:409:7)
    at async ModuleLoader.moduleProvider (node:internal/modules/esm/loader:291:45) {
  code: 'ERR_UNKNOWN_FILE_EXTENSION'
}
```

I hope this is fixed then too? Also great job here. All issues i have right now with your cms get fixed so quick its almost a miracle. :)

**Comment by jmikrut** [comment]
Hey @ckruppe of course! yes, the Local API as of right now imports lots of CSS and that gives Next.js a bad time in the pages router. The PR I linked completely resolves all CSS issues so you'll be golden across the board once it merges. 👍 Lots of other big performance improvements will be coming with that PR as well.

**Comment by ckruppe** [comment]
Hi @jmikrut sorry to bother you here on this Ticket again. I updated Payload today to the newest canary and also beta version but it seems the problems prevail. Is there anything i am missing? I share here my configs:

payload.config.ts
```typescript
import path from 'path';
import {fileURLToPath} from 'url';

import {mongooseAdapter} from '@payloadcms/db-mongodb';
import {slateEditor} from '@payloadcms/richtext-slate';
import {buildConfig} from 'payload';
import {de} from 'payload/i18n/de';
import sharp from 'sharp';

import {AdminUsers} from 'ServerConfigs/collections/AdminUsers';
import {Cards} from 'ServerConfigs/collections/Cards';
import {Recipients} from 'ServerConfigs/collections/Recipients';

const filename = fileURLToPath(import.meta.url);
const dirname = path.dirname(filename);

export default buildConfig({
    admin: {
        autoLogin: {
            email: 'c.kruppe@nfq.de',
            password: 'test',
            prefillOnly: true
        },
        livePreview: {
            collections: [Cards.slug],
            url: process.env.NEXT_PUBLIC_BASE_URL
        }
    },
    collections: [
        AdminUsers,
        Cards,
        Recipients
    ],
    db: mongooseAdapter({url: process.env.MONGODB_URI ?? ''}),
    editor: slateEditor({}),
    /**
     * Payload can now accept specific translations from 'payload/i18n/en'
     * This is completely optional and will default to English if not provided.
     */
    i18n: {supportedLanguages: {de}},
    plugins: [],
    secret: process.env.PAYLOAD_SECRET ?? '',
    // Sharp is now an optional dependency -
    // if you want to resize images, crop, set focal point, etc.
    // make sure to install it and pass it to the config.
    // This is temporary - we may make an adapter pattern
    // for this before reaching 3.0 stable
    sharp,
    typescript: {outputFile: path.resolve(dirname, 'types/payload-types.ts')}
});
```

next.config.js
```typescript
/* eslint-disable no-param-reassign */
const withBundleAnalyzer = require('@next/bundle-analyzer')({enabled: process.env.ANALYZE === 'true'});
const {withFeatureFlags} = require('@nfq/feature-flags/next');
const {WatchFontsConfigPlugin} = require('@nfq/next-fonts/webpack');
const {withPayload} = require('@payloadcms/next/withPayload');
const DuplicatePackageCheckerPlugin = require('duplicate-package-checker-webpack-plugin');

const liveFeatures = require('./features.live');
const stageFeatures = require('./features.stage');
const {permissions} = require('./headers');
const {withAliases} = require('./withAliases');
const withCors = require('./withCors');

/** @type {import('next').NextConfig} */
const nextConfig = {
    compiler: {styledComponents: true},
    eslint: {ignoreDuringBuilds: true},
    experimental: {
        swcPlugins: [[
            'nfq-feature-flags', {
                featureEnv: process.env.FEATURE_ENV || 'stage',
                featureFlags: {
                    live: liveFeatures,
                    stage: stageFeatures
                }
            }
        ]]
    },
    output: 'standalone',
    reactStrictMode: true,
    /**
     * The headers to send.
     *
     * @returns {Promise<Array<{headers: Array<{key: string, value: string}>, source: string}>>} The headers to send.
     */
    async headers() {
        return [{
            headers: [
                {
                    key: 'X-DNS-Prefetch-Control',
                    value: 'on'
                },
                {
                    key: 'Strict-Transport-Security',
                    value: 'max-age=63072000; includeSubDomains; preload'
                },
                {
                    key: 'X-XSS-Protection',
                    value: '1; mode=block'
                },
                {
                    key: 'X-Frame-Options',
                    value: 'SAMEORIGIN'
                },
                {
                    key: 'Referrer-Policy',
                    value: 'same-origin'
                },
                {
                    key: 'X-Content-Type-Options',
                    value: 'nosniff'
                },
                {
                    key: 'Permissions-Policy',
                    value: permissions.join(', ')
                }
            ],
            source: '/(.*)'
        }];
    },
    /**
     * The webpack config.
     *
     * @param {import('webpack').Configuration} config The config to use.
     * @returns {any} The nextjs webpack config changed.
     */
    webpack(config) {
        config.module.rules.push({
            test: /\.ttf|\.woff2/u,
            type: 'asset/resource'
        });

        config.plugins.push(new DuplicatePackageCheckerPlugin({
            /**
             * Excludes packages from checking.
             *
             * @param {object} instance The module instance object.
             * @returns {boolean} True if the package should be excluded.
             */
            exclude(instance) {
                return instance.name === 'react-is';
            }
        }));

        config.plugins.push(new WatchFontsConfigPlugin({
            globalStylesPath: './src/client/ui/utils/globalStyles.ts',
            outputPath: './src/client/ui/assets/fonts/fonts.css'
        }));

        return config;
    }
};

module.exports = phase => withBundleAnalyzer(
    withFeatureFlags(
        withAliases(
            withCors(
                withPayload(nextConfig),
                {phase}
            )
        ),
        {phase}
    )
);
```

And the simplified Controller class again too:
```typescript

import {type BotBuilderCloudAdapter, NotificationTargetType} from '@microsoft/teamsfx';
import {getPayload} from 'payload';

import configPromise from '@payload-config';

import {conversationBotApp} from 'Utils/conversationBotApp';

export class CardService {
    static async filterMembersByRecipients(members: BotBuilderCloudAdapter.Member[]) {
        const payload = await getPayload({config: configPromise});
        const recipients = await payload.find({
            collection: 'recipients',
            depth: 2,
            limit: 500
        });

        console.log(recipients);
        console.log(members);

        return [];
    }
}
```

The Error i am getting is again the same as before:
```
TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".scss" for C:\projekte\Intern\Pages\coding-cards\node_modules\@payloadcms\richtext-slate\dist\field\buttons.scss
    at Object.getFileProtocolModuleFormat [as file:] (node:internal/modules/esm/get_format:160:9)
    at defaultGetFormat (node:internal/modules/esm/get_format:203:36)
    at defaultLoad (node:internal/modules/esm/load:143:22)
    at async ModuleLoader.load (node:internal/modules/esm/loader:409:7)
    at async ModuleLoader.moduleProvider (node:internal/modules/esm/loader:291:45) {
  code: 'ERR_UNKNOWN_FILE_EXTENSION'
  ```
  
  Do i have to configure something differently?

**Comment by jmikrut** [comment]
Hey @ckruppe no problem at all! Happy to help.

So, we did release the new fixes as a canary but they are not yet released as a beta. That will be happening shortly. 

Did you have to migrate any of your code to match the Canary version that you installed? I want to make sure you got on the right version first.

Can you share your package.json? Make sure to `rm -Rf .next` in your folder as well.

**Comment by ckruppe** [comment]
Sure no problem here my Package json.

```
{
  "name": "coding-cards",
  "version": "1.0.0",
  "engines": {
    "node": ">=18.20.0"
  },
  "private": true,
  "standard-version": {
    "scripts": {
      "prebump": "nfqBump"
    }
  },
  "sideEffects": [
    "**/*.css"
  ],
  "template": {
    "version": "2.5.0",
    "name": "nextjs-typescript-template",
    "path": "https://github.com/nfqde/nextjs-typescript-template.git"
  },
  "scripts": {
    "dev:teamsfx:testtool": "env-cmd --silent -f .teamsTestTool/.localConfigs.testTool npm run dev",
    "dev:teamsfx:launch-testtool": "env-cmd --silent -f .env.testtool teamsapptester start",
    "analyze": "cross-env ANALYZE=true FEATURE_ENV=live next build",
    "analyze:browser": "cross-env ANALYZE=true BUNDLE_ANALYZE=browser FEATURE_ENV=live next build",
    "analyze:server": "cross-env ANALYZE=true BUNDLE_ANALYZE=server FEATURE_ENV=live next build",
    "build:live": "cross-env NODE_ENV=production FEATURE_ENV=live next build",
    "build:stage": "cross-env NODE_ENV=production FEATURE_ENV=stage next build",
    "build:test": "cross-env NODE_ENV=production FEATURE_ENV=test next build",
    "check:types": "tsc -p .",
    "deploy": "standard-version && git push --follow-tags origin main",
    "dev": "python -t -m webbrowser https://devcards.nfq.local && cross-env FEATURE_ENV=stage yarn next dev -p 3000",
    "lint": "eslint ./src --ext='.js' --ext='.jsx' --ext='.ts' --ext='.tsx'",
    "lint:report": "eslint --output-file eslint_report.json --format json --ext='.js, .jsx, .ts, .tsx' ./src",
    "merge:files": "mochawesome-merge \"./cypress/reports/mochawesome*.json\" > \"./cypress/reports/index.json\"",
    "start": "cross-env NODE_ENV=production next start -p 3000",
    "start:test": "start-server-and-test 'yarn run start' http://localhost:3000 'yarn run test'",
    "start:test:merge": "start-server-and-test 'yarn run start' http://localhost:3000 'yarn run test:merge'",
    "test:open": "cross-env NODE_ENV=test yarn cypress open",
    "test:cypress:ct": "cross-env NODE_ENV=test yarn cypress run --component && yarn merge:files",
    "test:cypress:integration": "cross-env NODE_ENV=test yarn cypress run",
    "test:merge": "yarn run test:cypress:integration && yarn run test:cypress:ct && yarn nfqReport",
    "test": "yarn run test:cypress:integration && yarn run test:cypress:ct",
    "generate:types": "payload generate:types && nfq-generate-types -f \"@app/features\" -j \"@nfq/feature-flags/jsx\" -i \"WithFeature\" -x \"WithoutFeature\" -e stage",
    "storybook": "storybook dev -p 6006",
    "build-storybook": "cross-env NODE_ENV=development FEATURE_ENV=stage storybook build",
    "postinstall": "patch-package"
  },
  "dependencies": {
    "@atelier-disko/payload-lexical-react-renderer": "^1.0.4",
    "@fluentui/react-components": "^9.54.10",
    "@microsoft/adaptivecards-tools": "^1.3.4",
    "@microsoft/teamsfx": "^2.3.2",
    "@nfq/autobind": "^1.1.3",
    "@nfq/next-fonts": "^1.5.0",
    "@nfq/react-grid": "^5.2.2",
    "@nfq/typed-next-api": "^1.5.3",
    "@payloadcms/db-mongodb": "3.0.0-canary.fb81f02",
    "@payloadcms/live-preview-react": "3.0.0-canary.fb81f02",
    "@payloadcms/next": "3.0.0-canary-f3eb5b2",
    "@payloadcms/plugin-seo": "3.0.0-canary.fb81f02",
    "@payloadcms/richtext-lexical": "3.0.0-canary.fb81f02",
    "@payloadcms/richtext-slate": "3.0.0-canary.fb81f02",
    "@payloadcms/translations": "3.0.0-canary.fb81f02",
    "@payloadcms/ui": "3.0.0-canary.fb81f02",
    "botbuilder": "^4.22.3",
    "env-cmd": "^10.1.0",
    "framer-motion": "^11.3.24",
    "mobx": "^6.13.1",
    "mobx-react-lite": "^4.0.7",
    "next": "^15.0.0-rc.0",
    "next-connect": "^1.0.0",
    "nfq-feature-flags": "file:.yalc/nfq-feature-flags",
    "payload": "3.0.0-canary.fb81f02",
    "react": "19.0.0-rc.0",
    "react-dom": "19.0.0-rc.0",
    "sharp": "^0.33.4",
    "styled-components": "^6.1.12",
    "swr": "^2.2.5"
  },
  "devDependencies": {
    "@axe-core/react": "^4.9.1",
    "@babel/core": "^7.25.2",
    "@babel/eslint-parser": "^7.25.1",
    "@babel/plugin-proposal-decorators": "^7.24.7",
    "@babel/plugin-transform-class-properties": "^7.24.7",
    "@babel/plugin-transform-class-static-block": "^7.24.7",
    "@babel/plugin-transform-typescript": "^7.25.2",
    "@babel/preset-env": "^7.25.3",
    "@babel/preset-react": "^7.24.7",
    "@babel/preset-typescript": "^7.24.7",
    "@cypress/code-coverage": "^3.12.45",
    "@microsoft/teams-app-test-tool": "^0.2.1",
    "@next/bundle-analyzer": "^15.0.0-rc.0",
    "@next/env": "^15.0.0-rc.0",
    "@nfq/eslint-config": "^3.2.1",
    "@nfq/eslint-plugin": "^0.8.2",
    "@nfq/feature-flags": "^3.1.0",
    "@nfq/storybook-github-issues": "^1.2.3",
    "@nfq/test-reporter": "^1.1.5",
    "@storybook/addon-essentials": "^8.2.9",
    "@storybook/addon-links": "^8.2.9",
    "@storybook/blocks": "^8.2.9",
    "@storybook/nextjs": "^8.2.9",
    "@storybook/react": "^8.2.9",
    "@storybook/testing-library": "^0.2.2",
    "@stryker-mutator/core": "^8.5.0",
    "@swc/core": "^1.7.10",
    "@total-typescript/ts-reset": "^0.5.1",
    "@types/cypress__code-coverage": "^3.10.2",
    "@types/node": "^22.2.0",
    "@types/react": "^18.3.3",
    "@types/react-dom": "^18.3.0",
    "@typescript-eslint/eslint-plugin": "^7.18.0",
    "@typescript-eslint/parser": "^7.18.0",
    "babel-loader": "^9.1.3",
    "babel-plugin-istanbul": "^7.0.0",
    "babel-plugin-styled-components": "^2.1.4",
    "babel-plugin-transform-react-remove-prop-types": "^0.4.24",
    "cross-env": "^7.0.3",
    "cypress": "^13.13.2",
    "cypress-real-events": "^1.13.0",
    "duplicate-package-checker-webpack-plugin": "^3.0.0",
    "eslint": "^8.57.0",
    "eslint-import-resolver-alias": "^1.1.2",
    "eslint-plugin-array-func": "^4.0.0",
    "eslint-plugin-better-styled-components": "^1.1.2",
    "eslint-plugin-import": "^2.29.1",
    "eslint-plugin-jsdoc": "^50.2.1",
    "eslint-plugin-jsx-a11y": "^6.9.0",
    "eslint-plugin-no-unsanitized": "^4.0.2",
    "eslint-plugin-node": "^11.1.0",
    "eslint-plugin-perf-standard": "^1.0.3",
    "eslint-plugin-promise": "^7.1.0",
    "eslint-plugin-react": "^7.35.0",
    "eslint-plugin-react-hooks": "^4.6.2",
    "eslint-plugin-react-hooks-ssr": "^0.1.5",
    "eslint-plugin-redos": "4.5.0-beta.2",
    "eslint-plugin-security": "^3.0.1",
    "eslint-plugin-sort-destructure-keys": "^2.0.0",
    "eslint-plugin-storybook": "^0.8.0",
    "graphql": "16.9.0",
    "html-webpack-plugin": "^5.6.0",
    "mocha": "^10.7.3",
    "mochawesome": "^7.1.3",
    "mochawesome-merge": "^4.3.0",
    "nyc": "^17.0.0",
    "onecolor": "^4.1.0",
    "patch-package": "^8.0.0",
    "postinstall-postinstall": "^2.1.0",
    "react-is": "19.0.0-rc.0",
    "sass": "^1.77.8",
    "sass-loader": "^16.0.0",
    "semver": "^7.6.3",
    "start-server-and-test": "^2.0.5",
    "storybook": "^8.2.9",
    "typescript": "^5.5.4",
    "webpack": "^5.93.0",
    "webpack-cli": "^5.1.4",
    "webpack-dev-server": "^5.0.4"
  },
  "resolutions": {
    "react": "19.0.0-rc.0",
    "react-dom": "19.0.0-rc.0",
    "postcss": "^8.4.31"
  }
}
```

I did delete the yarn.lock, node_modules and .next folder too. :)

**Comment by ckruppe** [comment]
What i did see the 3.0.0-canary-f3eb5b2 was the only one there the newest canary version did not match up exactly. Could it be the problem? Its the newest version i get suggested by npm.

**Comment by ckruppe** [comment]
Hey @jmikrut my problem got resolved with an switch to 3.0.0-beta.79. As i looked to check about 3.0.0-canary-f3eb5b2 i have seen that an new beta version is out. After i switched every package to the newest beta it worked flawlessly. :) I Thank you very much for your help.
